### ISSUE TITLE
checkCOSMICfile  update

### ISSUE THREAD

**Issue by robinjugas** [lgmgeo/AnnotSV#185]
Hi, 
I report the issue with adding COSMIC Cosmic_CompleteCNA_v98_GRCh37.tsv.gz

The names of columns in tsv.gz seem to change recently. 

I tried to solve it with replacing ^CNV_ID  with ^COSMIC_CNV_ID but I got another issue: 

...COSMIC configuration (June 15 2023 - 16:41)
                ...reading /mnt/ssd/ssd_1/conda_envs/robin_annotsv/share/AnnotSV/Annotations_Human/FtIncludedInSV/COSMIC/GRCh37/CosmicCompleteCNA.tsv.gz
can't read "chrom": no such variable
    while executing
"lappend L_texteToWrite($chrom) "$chrom\t$start\t$end\t$id\t$cnvtype""
    (procedure "checkCOSMICfile" line 37)
    invoked from within
"checkCOSMICfile"

Best regards, Robin

**Comment by lgmgeo** [comment]
Hi Robin,

Actually, I don't have COSMIC data (specific licence required).
Can you give me the CosmicCompleteCNA.tsv.gz header ?

So far the header line was:
```
CNV_ID  ID_GENE gene_name       ID_SAMPLE       ID_TUMOUR       Primary site    Site subtype 1  Site subtype 2  Site subtype 3  Primary histology       Histology subtype 1     Histology subtype 2     Histology subtype 3       SAMPLE_NAME     TOTAL_CN        MINOR_ALLELE    MUT_TYPE        ID_STUDY        GRCh    Chromosome:G_Start..G_Stop
```
This header started with "CNV_ID". And the following columns were also required: "MUT_TYPE", "Chromosome:G_Start..G_Stop"

If there is a new header, I can add a new patch to AnnotSV.

Best,

Véronique

**Comment by robinjugas** [comment]
Hi, thank you for kind response. 
The header should be :+1: 

`---------------------------
COSMIC Copy Number Variants
---------------------------
 All copy number variants from the current release in a tab separated table. For more information on copy number data, please see https://cancer.sanger.ac.uk/cosmic/help/cnv/overview. [ Cosmic_CompleteCNA_v98_GRCh37.tsv.gz ]

  File Description

[column number:label] Heading                            Description
--------------------------------------------------------------------------------------------------------
[1:A]                 COSMIC_CNV_ID                      A Copy number variant identifier (COSCNV) is used to identify the copy number variants within the file.
[2:B]                 COSMIC_GENE_ID                     A unique COSMIC gene identifier (COSG) is used to identify a gene within the file. This identifier can be used to retrieve additional Gene information from the Cosmic_Genes file.
[3:C]                 GENE_SYMBOL                        The gene name for which the data has been curated in COSMIC. In most cases this is the accepted HGNC identifier.
[4:D]                 COSMIC_SAMPLE_ID                   A unique COSMIC sample identifier (COSS) is used to identify a sample. This identifier can be used to retrieve additional Sample information from the Cosmic_Sample file.
[5:E]                 SAMPLE_NAME                        The sample name can be derived from a number of sources. In many cases it originates from the cell line name. Other sources include names assigned by the annotators, or an incremented number assigned during an anonymization process..
[6:F]                 COSMIC_PHENOTYPE_ID                A unique COSMIC identifier (COSO) for the classification. This identifier can be used to retrieve tissue and histology information from the classification file.
[7:G]                 TOTAL_CN                           The sum of the major and minor allele counts eg if ABB, total copy number = 3.
[8:H]                 MINOR_ALLELE                       The number of copies of the least frequent allele eg if ABB, minor allele = A ( 1 copy) and major allele = B ( 2 copies).
[9:I]                 MUT_TYPE                           Defined as Gain or Loss. For ICGC samples; as defined in the original data. For TCGA samples reanalysed with ASCAT -
                                                            * LOSS = average genome ploidy <= 2.7 AND total copy number = 0 OR average genome ploidy > 2.7 AND total copy number < ( average genome ploidy - 2.7 )
                                                            * GAIN = average genome ploidy <= 2.7 AND total copy number >= 5 OR average genome ploidy > 2.7  AND total copy number >= 9
[10:J]                COSMIC_STUDY_ID                    A unique COSMIC study identifier (COSU) is used to identify a study that have involved this copy number variation.
[11:K]                CHROMOSOME                         The chromosome location of a given copy number variant (1-22, X, Y or MT)
[12:L]                GENOME_START                       The start coordinate of a given copy number variant
[13:M]                GENOME_STOP                        The end coordinate of a given copy number variant`

**Comment by lgmgeo** [comment]
Please donwload and uncompressed the attached file:
[AnnotSV-cosmic.zip](https://github.com/lgmgeo/AnnotSV/files/11770692/AnnotSV-cosmic.zip)

Replace the $ANNOTSV/share/tcl/AnnotSV/AnnotSV-cosmic.tcl file with this one
Rerun AnnotSV

Does it work? (sorry, I can not test without the CosmicCompleteCNA.tsv.gz file)

**Comment by robinjugas** [comment]
Hi, it says:

```
    ...COSMIC configuration (June 19 2023 - 10:36)
                ...reading /mnt/ssd/ssd_1/conda_envs/robin_annotsv/share/AnnotSV/Annotations_Human/FtIncludedInSV/COSMIC/GRCh37/CosmicCompleteCNA.tsv.gz
can't read "i_stop": no such variable
    while executing
"lindex $Ls $i_stop"
    (procedure "checkCOSMICfile" line 44)
    invoked from within
"checkCOSMICfile"

```


which I corrected by modyfying line 70 :  set end   [lindex $Ls $i_end]   instead of previous  $i_stop 


But then I got: 

```
  ...creation of the COSMIC bed file in /mnt/ssd/ssd_1/conda_envs/robin_annotsv/share/AnnotSV/Annotations_Human/Users/GRCh37/F                   done (June 19 2023 - 10:40)
/mnt/ssd/ssd_1/conda_envs/robin_annotsv/share/AnnotSV/Annotations_Human/Users/GRCh37/FtIncludedInSV/CosmicCompleteCNA_GRCh37.bed: Unexpected        -> not using the associated annotations
expected integer but got ""
    while executing
"lrepeat $L_length """
    (procedure "checkBed" line 62)
    invoked from within
"checkBed $notFormattedFile [file dirname $notFormattedFile]"
    (procedure "checkUsersBED" line 16)
    invoked from within
"checkUsersBED"
    (file "/mnt/ssd/ssd_1/conda_envs/robin_annotsv/bin/AnnotSV" line 228)

```

The CosmicCompleteCNA_GRCh37.bed is basically empty., just the header.

**Comment by lgmgeo** [comment]
Could you send me the 20 first lines of the CosmicCompleteCNA.tsv.gz file? (veronique.geoffroy@inserm.fr)
It would be easier for me to code.

**Comment by robinjugas** [comment]
Thank you for your help. Problem solved.

**Comment by lgmgeo** [comment]
Bug fix will be released in the next version (v3.3.7)

**Comment by lgmgeo** [comment]
v3.3.7 posted.
